core/src/test/scala/org/apache/spark/SparkConfSuite.scala

```diff
@@ -403,6 +403,33 @@ class SparkConfSuite extends SparkFunSuite with LocalSparkContext with ResetSyst
       assert(thrown.getMessage.contains(key))
     }
   }
+
+  test("spark.task.resource.gpu.count should be multiple of spark.task.cpus") {
+    val conf = new SparkConf()
+    conf.validateSettings()
+
+    // no executor gpus
+    conf.set(GPUS_PER_TASK.key, "2")
+    intercept[SparkException] {
+      conf.validateSettings()
+    }
+
+    // gpus need to be multiple of executor
+    conf.set(GPUS_PER_TASK.key, "3")
+    conf.set(EXECUTOR_GPUS.key, "3")
+    conf.set(EXECUTOR_CORES.key, "4")
+    conf.set(CPUS_PER_TASK.key, "1")
```

Comment:
Based on the requirement at SparkConf, seems we don't allow to `set(GPUS_PER_TASK.key, "1")` for this setting, right? But I think it is a reasonable case.